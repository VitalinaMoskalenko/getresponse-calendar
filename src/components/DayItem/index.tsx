import React from "react";
import styled from "styled-components";
import { SmallBody } from "../Paragraphs";

type HeaderPropsType = {
  isSelected: boolean;
};

const Container = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  width: 100px;
  height: 80px;
  padding: 10px;
  border-bottom: 2px solid ${({ theme }) => theme.colors.darkGrey};
`;

const AccomplishedIcon = styled.img`
  height: 20px;
`;

type PropsType = {
  meal: string;
  isAccomplished?: boolean;
};

const DayItem = ({ meal, isAccomplished }: PropsType) => {
  return (
    <Container>
      <SmallBody>{meal}</SmallBody>
      {isAccomplished && (
        <AccomplishedIcon src="https://static.thenounproject.com/png/966940-200.png" />
      )}
    </Container>
  );
};

export default DayItem;
