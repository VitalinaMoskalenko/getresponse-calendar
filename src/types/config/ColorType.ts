export enum ColorType {
  white = "white",
  black = "black",
  lightGrey = "#e1e1e1",
  darkGrey = "#535353",
  green = "#b2c200",
  orange = "#ff801a",
}
