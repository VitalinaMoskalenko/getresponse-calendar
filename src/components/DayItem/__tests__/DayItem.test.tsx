import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import DayItem from "../";
import { theme } from "../../../config";

it("renders DayItem correctly", () => {
  const aayItemComponent = renderer
    .create(
      <ThemeProvider theme={theme}>
        <DayItem meal="meal" isAccomplished />
      </ThemeProvider>
    )
    .toJSON();

  expect(aayItemComponent).toMatchSnapshot();
});
