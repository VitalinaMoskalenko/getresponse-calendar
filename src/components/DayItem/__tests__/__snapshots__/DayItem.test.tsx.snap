// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DayItem correctly 1`] = `
<div
  className="sc-dlnjwi bfyrKl"
>
  <p
    className="sc-gtsrHT fsyTPD"
  >
    meal
  </p>
  <img
    className="sc-hKFxyN byCXgy"
    src="https://static.thenounproject.com/png/966940-200.png"
  />
</div>
`;
